tests/Integration/Behaviour/Features/Context/Domain/CustomerGroupFeatureContext.php

```diff
@@ -0,0 +1,134 @@
+
+class CustomerGroupFeatureContext extends AbstractDomainFeatureContext
+{
+    /**
+     * "When" steps perform actions, and some of them store the latest result
+     * in this variable so that "Then" action can check its content
+     *
+     * @var mixed
+     */
+    protected $latestResult;
+
+    /**
+     * Registry to keep track of created/edited customer groups using references
+     *
+     * @var int[]
+     */
+    protected $registry = [];
+
+    /**
+     * @When /^I create a Customer Group "(.+)" with the following details:$/
+     *
+     * @param string $customerGroupReference
+     * @param TableNode $table
+     *
+     * @throws Exception
+     */
+    public function createCustomerUsingCommand(string $customerGroupReference, TableNode $table)
+    {
+        $data = $this->localizeByRows($table);
+        $commandBus = $this->getCommandBus();
+
+        $command = new AddCustomerGroupCommand(
+            $data['name'],
+            $data['reduction'],
+            $data['priceDisplayMethod'],
+            $data['showPrice']
+        );
+
+        /** @var GroupId $id */
+        $id = $commandBus->handle($command);
+        $this->latestResult = $id->getValue();
+        $this->registry[$customerGroupReference] = $id->getValue();
+    }
+
+    /**
+     * @When /^I query Customer Group "(.+)" I should get a Customer Group with properties:$/
+     */
+    public function assertQueryCustomerProperties($customerGroupReference, TableNode $table)
+    {
+        $expectedData = $table->getRowsHash();
+
+        $queryBus = $this->getQueryBus();
+        /** @var EditableCustomerGroup $result */
+        $result = $queryBus->handle(new GetCustomerGroupForEditing($this->registry[$customerGroupReference]));
+
+        $serializer = CommonFeatureContext::getContainer()->get('serializer');
+        $realData = $serializer->normalize($result);
+
+        DataComparator::assertDataSetsAreIdentical($expectedData, $realData);
+
+        $this->latestResult = null;
+    }
+
+    /**
+     * @Then customer group :productReference localized :fieldName should be:
+     *
+     * localizedValues transformation handled by @see LocalizedArrayTransformContext
+     *
+     * @param string $customerGroupReference
+     * @param string $fieldName
+     * @param array $expectedLocalizedValues
+     */
+    public function assertLocalizedPropertyForCustomerGroup(string $customerGroupReference, string $fieldName, array $expectedLocalizedValues): void
+    {
+        $queryBus = $this->getQueryBus();
+        /** @var EditableCustomerGroup $result */
+        $result = $queryBus->handle(new GetCustomerGroupForEditing($this->registry[$customerGroupReference]));
+
+        $this->assertLocalizedProperty($result, $fieldName, $expectedLocalizedValues);
+    }
+
+    private function assertLocalizedProperty(EditableCustomerGroup $editableCustomerGroup, string $fieldName, array $expectedLocalizedValues): void
```

Comment:
add docblock to explain $expectedLocalizedValues array in psalm notation

Reply:
there is identical (or very similar) method in `CombinationAssertionFeatureContext:assertLocalizedProperty`, I think you could move that method to AbstractDomainFeatureContext, make it protected and reuse instead of reimplementing it here.